### PR TITLE
Add test for 12.4.21 of Temporal

### PR DESCRIPTION
**Pull request by FrankYFTang**
12.4.21 Temporal.Calendar.prototype.fields ( fields )
https://tc39.es/proposal-temporal/#sec-temporal.calendar.prototype.fields

@sffc @Ms2ger @ryzokuken @ljharb

### CODE REVIEW

File: test/built-ins/Temporal/Calendar/fields.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+let cal = new Temporal.Calendar("iso8601")
+const fields = {
+  *[Symbol.iterator]() {
+      let i = 0;
+      while (i++ < 1000001) {
+        yield "garbage " + i;
+      }
+  }
+}
+assert.deepEqual(cal.fields(fields), Array.from(fields));
```
Comment (Ms2ger) [inline test/built-ins/Temporal/Calendar/fields.js:27]: It doesn't seem like `deepEqual` actually checks the result is an Array. Not sure why we'd use it over `compareArray`

File: test/built-ins/Temporal/Calendar/fields.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+// Copyright (C) 2021 the V8 project authors. All rights reserved.
```
Comment (Ms2ger) [inline test/built-ins/Temporal/Calendar/fields.js:1]: This should go in `test/built-ins/Temporal/Calendar/prototype/fields/`, but compare with the existing test https://github.com/tc39/proposal-temporal/blob/e057057bf18f81b226cd0d6b51faff3ac7da6ce7/polyfill/test/Calendar/prototype/fields/argument-iterable-not-array.js

File: test/built-ins/Temporal/Calendar/fields.js (hunk 1 above)
Comment (FrankYFTang) [reply test/built-ins/Temporal/Calendar/fields.js]: DONE

File: test/built-ins/Temporal/Calendar/fields.js (hunk 2 above)
Comment (FrankYFTang) [reply test/built-ins/Temporal/Calendar/fields.js]: DONE

**Comment by rwaldron** [comment]
@FrankYFTang you should make sure you coordinate with @jugglinmike, as I know that he's working on migrating and expanding all of the Temporal tests

**Comment by FrankYFTang** [comment]
> @FrankYFTang you should make sure you coordinate with @jugglinmike, as I know that he's working on migrating and expanding all of the Temporal tests

sure @jugglinmike do you think this test faithfully test against the current Temporal spec text?

**Review by jugglinmike — APPROVED** [review]
Yup, it looks accurate to me! Thanks for writing it!

(My guess is that @rwaldron was referring to [this issue I created to coordinate test-writing for the Temporal Proposal](https://github.com/tc39/test262/issues/3002). If you'd like to write more tests, please do let us know there. We still have more tests to write for this method, so I'll leave it as "unassigned" in that issue.)

**Pull request merged**